Documentation/contributing/docs/docsstructure.rst

```diff
@@ -0,0 +1,98 @@
+your feature. Instead, **document how your feature guides users on their
+journey.**
+
+Maintaining Good Information Architecture
+-----------------------------------------
+
+When you add, update, or remove documentation, consider how the change affects
+the site's `information architecture`_. Information architecture is what shapes
+a user's experience and their ability to accomplish their goals with Cilium. If
+an addition, change, or removal would significantly alter a user's journey or
+prevent their success, make sure to flag the change clearly in :ref:`upgrade
+notes <current_release_required_changes>`.
+
+.. _information architecture: https://www.usability.gov/what-and-why/information-architecture.html
+
+Adding a New Page
+-----------------
+
+When you need to write completely new content, create one or more new pages as
+one of the three following types:
+
+- Concept (no steps, just knowledge)
+- Task (how to do one discrete thing)
+- Tutorial (how to combine multiple features to accomplish specific goals)
+
+A *concept* explains some aspect of Cilium. Typically, concept pages don't
+include sequences of steps. Instead, they link to tasks or tutorials.
+
+For an example of a concept page, see :ref:`Routing <Routing>`.
+
+A *task* shows how to do one discrete thing with Cilium. Task pages give
+readers a sequence of steps to perform. A task page can be short or long, but
+must remain focused on the task's singular goal. Task pages can blend brief
+explanations with the steps to perform, but if you need to provide a lengthy
+explanation, write a separate concept and link to it. Link related task and
+concept pages to each other.
+
+For an example of a task page, see :ref:`Migrating a Cluster to Cilium
+<cni_migration>`.
+
+A *tutorial* shows how to accomplish a goal using multiple Cilium features.
+Tutorials are flexible: for example, a tutorial page could provide several
+discrete sequences of steps to perform, or show how related pieces of code
+could interact. Tutorials can blend brief explanations with the steps to
+perform, but lengthy explanations should link to related concept topics.
+
+For an example of a tutorial page, see :ref:`Inspecting Network Flows with the
+CLI <hubble_cli>`.
+
+.. note::
+
+  You may need to add multiple pages to support a new feature. For example, if
+  a new feature requires an explanation of its underlying ideas, add a concept
+  page as well as a task page.
+
+Updating an Existing Page
+-------------------------
+
+Consider whether you can update an existing page or whether to add a new one.
+
+If adding or updating content to a page keeps it centered on a single concept
+or task, then you can update an existing page. If adding or updating content to
+a page expands it to include multiple concepts or tasks, then add new pages for
+individual concepts and tasks.
+
+If you're moving a page and changing its URL, make sure you update every link
```

Comment:
Should this also mention the need to potentially update the link map from newer version to older versions?

Reply:
Do you mean the redirects?

Contributors don't usually have access to the relevant config (I think it's in the ReadTheDocs interface), but we could suggest asking on Slack if necessary

Reply:
Yeah that's what I mean. That's a good point; yeah maybe asking on slack is the best option.

Reply:
OK, I added it. Thanks!